### PR TITLE
[argparse] Namespace

### PR DESCRIPTION
**Pull request by tlaziuk**
add type definitions for Namespace

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/nodeca/argparse/blob/1.0.9/lib/namespace.js>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by tlaziuk** [comment]
`lint` prints a lot of errors, I'll try to fix them

**Comment by tlaziuk** [comment]
fixing the lint is done, now I'm waiting for review and approval

**Review by arcticwaters — APPROVED** [review]
Would have preferred less whitespace changes, but lgtm

**Pull request merged**
